rules/opentitan.bzl

```diff
@@ -71,6 +72,147 @@ obj_transform = rv_rule(
     toolchains = ["@rules_cc//cc:toolchain_type"],
 )
 
+# A provider for device-specific archive files that hold binaries of SRAM programs.
+ArchiveInfo = provider(fields = ["archive_infos"])
+
+def _bin_to_archive_impl(ctx):
+    cc_infos = []
+    cc_toolchain = find_cc_toolchain(ctx).cc
+    cc_info_dict = {}
+    for (device, binary_target) in zip(ctx.attr.devices, ctx.attr.binaries):
```

Comment:
Nit: Is it possible to assert that the two lists have the same length? Typically, `zip(a, b)` yields `min(len(a), len(b))` elements, but I'm not sure about the semantics in Bazel.